### PR TITLE
Kepcotrend minor bugfix and prettier

### PR DESCRIPTION
**Pull request by christinahedges**
Following changes have been added:

* kepcotrend and keprange have been slightly improved with prettier plot colors. The range of masked data is also plotted in kepcotrend. 
* kepcotrend has had a minor bug fixed (a single cotrending basis vector can now be used)
* kepcotrend has had the keyword 'non-interactive' added (so plot windows can be suppressed.) 
* kepcotrend now saves the plot in the same style as other routines (appending '.png' to the file name.)

I will update the changelog to reflect this. 
@mirca what do you think? See any blank lines?

### CODE REVIEW

File: pyke/kepcotrend.py [hunk 1]
```diff
@@ -339,22 +341,30 @@ def do_plot(date, flux_old, flux_new, bvsum, cad, good_data, cad_nans, version):
...
         plot_y = bvsum_sub[mask]
-        plt.plot(plot_x, plot_y, color='red', linestyle='-', linewidth=2.0)
+        plt.plot(plot_x, plot_y, color='#c0392b', linestyle='-', linewidth=2.0)
     date2 = np.insert(date_sub, [0], [date_sub[0]])
     date2 = np.append(date2, [date_sub[-1]])
     flux2 = np.insert(flux_old_sub,[0], [0.0])
     flux2 = np.append(flux2, [0.0])
-    plt.fill(date2, flux2, fc='#FFFACD', linewidth=0.0)
+    plt.fill(date2, flux2, color='#a8a7a7', linewidth=0.0, alpha=0.2, label='Data')
+    if (maskdata is None) is False:
```
Comment (mirca) [inline pyke/kepcotrend.py]: That's funny, I'm pretty sure you can do:
```
if maskdata:
```

File: pyke/kepcotrend.py [hunk 2]
```diff
@@ -339,22 +341,30 @@ def do_plot(date, flux_old, flux_new, bvsum, cad, good_data, cad_nans, version):
...
+        for m in maskdata:
+            pos = np.where((barytime0 + 2400000 + date2 >m[0]) & (barytime0 + 2400000 + date2 <=m [1]))[0]
+            plt.fill_between(date2[pos], flux2[pos].min(), flux2[pos].max(), color='#c0392b', linewidth=0.0, alpha=0.3, label='Masked')
+
+
```
Comment (mirca) [inline pyke/kepcotrend.py]: rm blank line

File: pyke/kepcotrend.py [hunk 3]
```diff
@@ -339,22 +341,30 @@ def do_plot(date, flux_old, flux_new, bvsum, cad, good_data, cad_nans, version):
...
     if ymin1 - yr1 * 0.01 <= 0.0:
         plt.ylim(1.0e-10, ymax1 + yr1 * 0.01)
     else:
         plt.ylim(ymin1 - yr1 * 0.01, ymax1 + yr1 * 0.01)
     plt.xlabel(xlab, {'color' : 'k'})
     plt.ylabel(ylab1, {'color' : 'k'})
-    plt.grid()
+    plt.grid(ls='--',alpha=0.3)
+    plt.legend()
+
```
Comment (mirca) [inline pyke/kepcotrend.py]: rm blank line

File: pyke/kepcotrend.py [hunk 4]
```diff
@@ -367,14 +377,18 @@ def do_plot(date, flux_old, flux_new, bvsum, cad, good_data, cad_nans, version):
...
 
     date2 = np.insert(date_sub, [0], [date_sub[0]])
     date2 = np.append(date2, [date_sub[-1]])
     flux2 = np.insert(flux_new_sub, [0], [0.0])
     flux2 = np.append(flux2, [0.0])
-    plt.fill(date2, flux2, fc='#FFFACD', linewidth=0.0)
+    plt.fill(date2, flux2, fc='#a8a7a7', alpha=0.2,  linewidth=0.0)
+    if (maskdata is None) is False:
+        for m in maskdata:
+            pos = np.where((barytime0 + 2400000 + date2 >m[0]) & (barytime0 + 2400000 + date2 <=m [1]))[0]
```
Comment (mirca) [inline pyke/kepcotrend.py]: add a white space after ``>`` and ``<=``

File: pyke/kepcotrend.py [hunk 5]
```diff
@@ -384,14 +398,15 @@ def do_plot(date, flux_old, flux_new, bvsum, cad, good_data, cad_nans, version):
 
     plt.xlabel(xlab, {'color' : 'k'})
     plt.ylabel(ylab2, {'color' : 'k'})
-    plt.grid()
+    plt.grid(ls='--',alpha=0.3)
```
Comment (mirca) [inline pyke/kepcotrend.py]: white space after comma

File: pyke/kepcotrend.py (hunk 1 above)
Comment (christinahedges) [reply pyke/kepcotrend.py]: You're right. Doy.

File: pyke/kepcotrend.py (hunk 2 above)
Comment (christinahedges) [reply pyke/kepcotrend.py]: blank lines ftw

File: pyke/kepcotrend.py (hunk 3 above)
Comment (christinahedges) [reply pyke/kepcotrend.py]: blank lines 4eva

**Comment by christinahedges** [comment]
Updated a unit test for you @mirca

File: pyke/kepcotrend.py [hunk 6]
```diff
@@ -367,14 +375,18 @@ def do_plot(date, flux_old, flux_new, bvsum, cad, good_data, cad_nans, version):
...
+        plt.scatter(plot_x, plot_y, color='#363636', linestyle='-', linewidth=1.0, marker='.', s=5)
         plot_y = bvsum_sub[mask]
 
     date2 = np.insert(date_sub, [0], [date_sub[0]])
     date2 = np.append(date2, [date_sub[-1]])
     flux2 = np.insert(flux_new_sub, [0], [0.0])
     flux2 = np.append(flux2, [0.0])
-    plt.fill(date2, flux2, fc='#FFFACD', linewidth=0.0)
+    plt.fill(date2, flux2, fc='#a8a7a7', alpha=0.2,  linewidth=0.0)
+    if (maskdata is None) is False:
```
Comment (mirca) [inline pyke/kepcotrend.py]: same thing here:
```
if maskdata:
```

File: pyke/kepcotrend.py [hunk 7]
```diff
@@ -339,22 +341,28 @@ def do_plot(date, flux_old, flux_new, bvsum, cad, good_data, cad_nans, version):
...
         plot_y = bvsum_sub[mask]
-        plt.plot(plot_x, plot_y, color='red', linestyle='-', linewidth=2.0)
+        plt.plot(plot_x, plot_y, color='#c0392b', linestyle='-', linewidth=2.0)
     date2 = np.insert(date_sub, [0], [date_sub[0]])
     date2 = np.append(date2, [date_sub[-1]])
     flux2 = np.insert(flux_old_sub,[0], [0.0])
     flux2 = np.append(flux2, [0.0])
-    plt.fill(date2, flux2, fc='#FFFACD', linewidth=0.0)
+    plt.fill(date2, flux2, color='#a8a7a7', linewidth=0.0, alpha=0.2, label='Data')
+    if maskdata is not None:
```
Comment (mirca) [inline pyke/kepcotrend.py]: that looks good :)

**Comment by barentsen** [comment]
:+1: Thanks @christinahedges and @mirca 

If you merge 2 more pull requests, you will get the surprise that is in this box:  🎁

**Pull request merged**